modules/platforms/dotnet/Apache.Ignite.Core/Impl/PlatformJniTarget.cs

```diff
@@ -177,6 +177,9 @@ public T OutStream<T>(int type, Func<IBinaryStream, T> readAction)
             }
             catch (JavaException jex)
             {
+                if (errorAction != null)
+                    return errorAction.Invoke(jex);
```

Comment:
I think this should be `return errorAction.Invoke(ConvertException(jex));`. Otherwise, when `errorAction` rethrows the exception, it is not converted properly.

Reply:
Fixed